### PR TITLE
Add featured events and study group styles; fix card inconsistencies

### PR DESCRIPTION
**Pull request by lilyndk**
## Description
- updated homepage announcements to work for both featured announcements and events
-- if it's not an event, the announcement will have default white background and grey border styles
-- added second meetup event; added square event teaser images
- added featured study group styles

Misc. additions:
- created icon size variable
- added missing default font weight '300'
- added missing rounded corners to images with decreased border-radius size
- fixed misc. card inconsistencies and hardcoded values
- fixed inconsistent nav font sizes
- added page intro to 'Mentorship resources'

## Change Type
- [ ] Bug Fix
- [X] New Feature
- [X] Code Refactor
- [ ] Mentor Update
- [ ] Documentation
- [ ] Other

## Related Issue
Events: https://github.com/WomenCodingCommunity/WomenCodingCommunity.github.io/issues/192
Active study groups: https://github.com/WomenCodingCommunity/WomenCodingCommunity.github.io/issues/121

## Screenshots
AFTER:
screencapture-127-0-0-1-4000-2024-05-15-22_36_57
BEFORE:
screencapture-womencodingcommunity-2024-05-15-20_18_39

AFTER:
screencapture-127-0-0-1-4000-program-study-group-2024-05-15-22_38_16
BEFORE:
screencapture-womencodingcommunity-program-study-group-2024-05-15-20_26_39

AFTER:
screencapture-127-0-0-1-4000-mentorship-resources-2024-05-15-20_23_59
BEFORE:
screencapture-womencodingcommunity-mentorship-resources-2024-05-15-20_24_12

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [X] I checked and followed the [contributor guide](CONTRIBUTING.md) 
- [X] I have tested my changes locally.
- [X] I have added a screenshot from the website after I tested it locally

### CODE REVIEW

File: _data/events.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+- title: |
+    Book Club Online Event: The Pragmatic Programmer
```
Comment (dricazenck) [inline _data/events.yml]:
```
    Online Book Club
```
or 

```
     Book Club: The Pragmatic Programmer
```

File: _data/events.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  time: 8:00 PM to 9:30 PM CEST
+  image:
+    path: "/assets/images/events/event-book-club-053024.jpeg"
+    alt: The Pragmatic Programmer book cover
+  link:
+    path: https://www.meetup.com/women-coding-community/events/300856519
+    title: View meetup event
+    target: _blank
+- title: |
+    Getting Started with IaC - An Introduction To Terraform
```
Comment (dricazenck) [inline _data/events.yml]:
```suggestion
    Getting Started with IaC
```

File: _includes/featured-announcements.html [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<div class="container announcements">
+    <div class="row featured-content">
+        <div class="row-header">
+            <h2>Announcements</h2>
```
Comment (dricazenck) [inline _includes/featured-announcements.html:4]: I think it is missing this screenshot.. how it is with the layout changes.

**Review by dricazenck — APPROVED** [review]

File: _includes/featured-announcements.html [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+                                <h3 class="card-title">{{ announcement.title }}</h3>
+                            </div>
+                            <div class="card-media col-12 col-sm-3 col-lg-5">
+                                <img src="{{ announcement.image.path }}" alt="{{ announcement.image.alt }}">
+                            </div>
+                        </div>
+                        <div class="card-body col-12">
+                            <p>{{ announcement.description }}</p>
+                            {% if announcement.link %}
+                                <a href="{{ announcement.link.path }}" target="{{ announcement.link.target }}" class="btn btn-outline-primary">{{announcement.link.title }}</a>
```
Comment (dricazenck) [inline _includes/featured-announcements.html]: Could we align this bottom to the bottom?

File: _data/events.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
+- title: |
```
Comment (dricazenck) [inline _data/events.yml:1]: I think this event it is also in announcements, maybe is better to remove from there to not be duplicated in home page.

File: program-study-group.html [hunk 6]
```diff
@@ -8,39 +8,38 @@
...
-                <p>Our study groups cover a variety of topics and are designed to accommodate different schedules and
-                    experience
-                    levels. Whether you're a beginner looking to dive into a new technology or an experienced professional
-                    seeking
-                    to expand your expertise, there's a study group for you.</p>
-                <p>These study groups typically last anywhere from 1-2 months to longer durations, allowing participants to
-                    delve
-                    deeply into the subject matter and benefit from the guidance of experienced mentors.</p>                    
-            </section>
+                <p>Explore our technical study groups, created in collaboration with available mentors, to deepen your knowledge and skills in specific areas of interest.</p>
```
Comment (dricazenck) [inline program-study-group.html:11]: Could change the title to be `Study Groups` It is not a programme.

File: program-study-group.html [hunk 7]
```diff
@@ -8,39 +8,38 @@
...
+                <p>Our study groups cover a variety of topics and are designed to accommodate different schedules and experience levels. Whether you're a beginner looking to dive into a new technology or an experienced professional seeking to expand your expertise, there's a study group for you.</p>
+                <p>These study groups typically last anywhere from 1-2 months to longer durations, allowing participants to delve deeply into the subject matter and benefit from the guidance of experienced mentors.</p>                    
+                <p>Join our <a href="https://womencodingcommunity.slack.com/archives/C073W920RND" target="_blank"> Study Group Slack Channel</a>.</p>
```
Comment (dricazenck) [inline program-study-group.html:14]:
```suggestion
                <p>Join our <a href="https://womencodingcommunity.slack.com/archives/C073W920RND" target="_blank"> Study Group Slack Channel</a>.</p>
```

**Comment by lilyndk** [comment]
@dricazenck screenshot of announcement styles: 
Screenshot 2024-05-15 at 22 51 07

**Pull request merged**